drivers/cc1200/cc1200-rxtx.c

```diff
@@ -0,0 +1,430 @@
+                 * Don't know, why this waiting is necessary, but the driver
+                 * blocks without this. Alternatively, the delay can also be
+                 * located in  _netdev_cc1200_isr() or _isr(), but this here
+                 * is more specific. Also, the delay can be placed before or
+                 * after, but not in _rx_start()...
+                 */
+                xtimer_spin(xtimer_ticks_from_usec(800));
+            }
+            else {
+                DEBUG("cc1200_isr_handler((): isr handled too slow or falling edge trigger\n");
+                _rx_abort(dev);
+            }
+            break;
+        case RADIO_RX_BUSY:
+            //DEBUG("radio rx busy\n");
+            _rx_continue(dev, callback, arg);
+            break;
+        case RADIO_TX_BUSY:
+            //DEBUG("radio tx busy\n");
+            if (!gpio_read(dev->params.int_pin)) {
+                _tx_continue(dev);
+            }
+            else {
+                //DEBUG("cc1200_isr_handler() RADIO_TX_BUSY + GDO2\n");
+            }
+            break;
+        default:
+            DEBUG("%s:%s:%u: unhandled mode\n", RIOT_FILE_RELATIVE,
+                  __func__, __LINE__);
+            break;
+    }
+}
+
+int cc1200_send(cc1200_t *dev, const struct iovec *vector, unsigned count)
+{
+    DEBUG("%s:%u\n", __func__, __LINE__);
+
+    switch (dev->radio_state) {
+        case RADIO_RX_BUSY:
+            DEBUG("%s: rx busy...\n", __func__);
+            return -EAGAIN;
+        case RADIO_TX_BUSY:
+            DEBUG("%s: tx busy...\n", __func__);
+            //return -EAGAIN;
+            {
+                int tx_bytes = cc1200_get_reg_robust(dev, CC1200_NUM_TXBYTES);
+
+                while (tx_bytes > 0) {
+                    DEBUG("%s: waiting for transmission to be finished, %d bytes left\n",
+                          __func__, tx_bytes);
+                    thread_yield();
+                    xtimer_spin(xtimer_ticks_from_usec(50));
+                    tx_bytes = cc1200_get_reg_robust(dev, CC1200_NUM_TXBYTES);
+                }
+
+                /*
+                 * need to wait a little before sending next frame;
+                 * don't know why...
+                 *
+                 * TODO: check in detail...
+                 */
+                xtimer_usleep(500);
+            }
+    }
+
+    gpio_irq_disable(dev->params.int_pin);
+    dev->radio_state = RADIO_TX_BUSY;
+
+    /*
+     * Number of bytes to send is:
+     * length of phy payload (packet->length)
+     * + size of length field (1 byte)
+     */
+    //size = packet->length + 1;
+    uint8_t size = 0;
+    for (uint8_t i = 0; i < count; i++) size += vector[i].iov_len;
+
+    DEBUG("%s: size=%d\n", __func__, size);
+
+    if (size > (CC1200_PACKET_LENGTH + 1)) {
+        DEBUG("%s:%s:%u trying to send oversized packet\n",
+              RIOT_FILE_RELATIVE, __func__, __LINE__);
+        return -ENOSPC;
+    }
+
+#ifdef MODULE_CC1200_HOOKS
```

Comment:
Even though I don't know what this module intends to do. Where is it defined? At lease I would have expected a pseudomodule or something

Reply:
Oh yeah I am sorry. That is something I took over from the CC110X, as it seemed handy in case I needed to implement alternative functions. I will remove those, as they are not needed.